src/modules/land_detector/LandDetector.cpp

```diff
@@ -120,22 +131,27 @@ void LandDetector::Run()
 		_landDetected.in_ground_effect = in_ground_effect;
 
 		int instance;
-		orb_publish_auto(ORB_ID(vehicle_land_detected), &_landDetectedPub, &_landDetected,
+		orb_publish_auto(ORB_ID(vehicle_land_detected), &_land_detected_pub, &_landDetected,
 				 &instance, ORB_PRIO_DEFAULT);
 	}
 
 	// set the flight time when disarming (not necessarily when landed, because all param changes should
 	// happen on the same event and it's better to set/save params while not in armed state)
-	if (_takeoff_time != 0 && !_arming.armed && _previous_arming_state) {
+	if (_takeoff_time != 0 && !_arming.armed && _previous_armed_state) {
 		_total_flight_time += now - _takeoff_time;
 		_takeoff_time = 0;
+
 		uint32_t flight_time = (_total_flight_time >> 32) & 0xffffffff;
-		param_set_no_notification(_p_total_flight_time_high, &flight_time);
+		_param_total_flight_time_high = flight_time;
```

Comment:
@bkueng, I don't quite understand how to update the value of a param under this paradigm. Would you point me to an example? Thanks!